utPlayer.py

```diff
@@ -31,42 +31,64 @@
 
 from framework.core.logModule import logModule
 from framework.plugins.ut_raft.interactiveShell import InteractiveShell
+from framework.plugins.ut_raft.configRead import ConfigRead
+
 class utPlayer():
     """
     UT Player class
     """
-    def __init__(self, session:object, playerTool:dict={'tool': 'gstreamer', 'prerequisites':[]}, log:logModule=None):
+    playerConfig = os.path.join(dir_path, "uPlayerConfig.yml")
+
+    def __init__(self, session:object, platform:str, player:str = "gstreamer", log:logModule=None):
         """
         Initializes player class.
 
         Args:
             session (class): The session object to communicate with the device
+            platform (str): Platfrom name
+            player (str, optional) : Player to use. Defaults to gstreamer
             log (class, optional): Parent log class. Defaults to None.
         """
         self.log = log
         if log is None:
             self.log = logModule(self.__class__.__name__)
             self.log.setLevel( self.log.INFO )
         self.session = session
-        self.playbackTool = playerTool["tool"]
-        for cmd in playerTool["prerequisites"]:
-            self.session.write(cmd)
+        self.playerProfile = ConfigRead(self.playerConfig, platform)
+        self.player = self.playerProfile.get(player)
```

Comment:
this is odd. Whilst you can do this you don't have too and not sure why you would want too?

```python
self.player = self.playerProfile.player
```

although it's the same as this, it's not the reason why the class exists.

```python
self.player = self.playerProfile.get(player)
```

Reply:
In this case, player is a variable and not part of the configuration file. Therefore, self.playerProfile.player won't work, which is why the get function is used.